src/wasm/wasm-type.cpp

```diff
@@ -227,8 +265,46 @@ template<typename Self> struct TypeGraphWalkerBase {
   // This base walker does not know when to stop scanning, so at least one of
   // these needs to be overridden with a method that calls the base scanning
   // method only if some end condition isn't met.
-  void scanType(Type* type);
-  void scanHeapType(HeapType* ht);
+  void scanType(Type* type) {
+    if (type->isBasic() || type->isRef()) {
+      return;
+    }
```

Comment:
What do these three lines do? The early return doesn't seem needed since `isTuple()` is disjoint from them?

Reply:
These could be replaced with a comment calling out that basic types and reference types are deliberately not handled.